dvc/api.py

```diff
@@ -110,10 +110,13 @@ def read(path, repo=None, rev=None, remote=None, mode="r", encoding=None):
 @contextmanager
 def _make_repo(repo_url, rev=None):
     if rev is None and (not repo_url or os.path.exists(repo_url)):
-        yield Repo(repo_url)
-    else:
-        with external_repo(url=repo_url, rev=rev) as repo:
-            yield repo
+        try:
+            yield Repo(repo_url)
+            return
+        except NotDvcRepoError:
+            pass  # fallthrough to external_repo
+    with external_repo(url=repo_url, rev=rev) as repo:
```

Comment:
This will fails if `repo_url` is `None`, which means cur dir really.

Reply:
@Suor Is there a test for that? I don't remember us discussing `None` here at all.

Reply:
In all api funcs, `repo` defaults to `None`, which means current dir, same for `Repo()`.

Reply:
Fixed by cdd9d0d78488daf159e1d0847a421d4929f44d61